pkg/providers/instancetype/instancetype.go

```diff
@@ -93,13 +97,106 @@ func (p *DefaultProvider) List(ctx context.Context, kc *v1alpha1.KubeletConfigur
+		RegionId:            tea.String(p.region),
+		DestinationResource: tea.String("InstanceType"),
+	}
+	runtime := &util.RuntimeOptions{}
 
-	// TODO: implement me
+	resp, err := p.ecsClient.DescribeAvailableResourceWithOptions(describeAvailableResourceRequest, runtime)
+	if err != nil {
+		return err
+	}
+
+	if resp.Body == nil || resp.Body.AvailableZones == nil || len(resp.Body.AvailableZones.AvailableZone) == 0 {
+		return errors.New("DescribeAvailableResourceWithOptions failed to return any instance types")
+	}
+
+	for _, az := range resp.Body.AvailableZones.AvailableZone {
+		if *az.Status == "Available" { // Available or SoldOut
+			for _, ar := range az.AvailableResources.AvailableResource {
+				for _, sr := range ar.SupportedResources.SupportedResource {
+					if *sr.Status == "Available" { // Available or SoldOut
+						if _, ok := instanceTypeOfferings[*sr.Value]; !ok {
+							instanceTypeOfferings[*sr.Value] = sets.New[string]()
+						}
+						instanceTypeOfferings[*sr.Value].Insert(*az.ZoneId)
+					}
+				}
+			}
+		}
+	}
+
+	if p.cm.HasChanged("instance-type-offering", instanceTypeOfferings) {
+		// Only update instanceTypesSeqNun with the instance type offerings  have been changed
+		// This is to not create new keys with duplicate instance type offerings option
+		atomic.AddUint64(&p.instanceTypeOfferingsSeqNum, 1)
+		log.FromContext(ctx).WithValues("instance-type-count", len(instanceTypeOfferings)).V(1).Info("discovered offerings for instance types")
+	}
+	p.instanceTypeOfferings = instanceTypeOfferings
 	return nil
 }
+
+func getAllInstanceTypes(client *ecsclient.Client) ([]*ecsclient.DescribeInstanceTypesResponseBodyInstanceTypesInstanceType, error) {
+	var InstanceTypes []*ecsclient.DescribeInstanceTypesResponseBodyInstanceTypesInstanceType
+
+	describeInstanceTypesRequest := &ecsclient.DescribeInstanceTypesRequest{
+		/*
+			Reference: https://api.aliyun.com/api/Ecs/2014-05-26/DescribeInstanceTypes caveat:
+			The maximum value of Max Results (maximum number of entries per page) parameter is 100,
+			for users who have called this API in 2022, the maximum value of Max Results parameter is still 1600,
+			on and after November 15, 2023, we will reduce the maximum value of Max Results parameter to 100 for all users,
+			and no longer support 1600, if If you do not pass the Next Token parameter for paging when you call this API,
+			only the first page of the specification (no more than 100 items) will be returned by default.
+		*/
+		MaxResults: tea.Int64(100),
+	}
+	runtime := &util.RuntimeOptions{}
+
+	for {
+		resp, err := client.DescribeInstanceTypesWithOptions(describeInstanceTypesRequest, runtime)
+		if err != nil {
+			return nil, err
+		}
+
+		if resp.Body == nil || resp.Body.InstanceTypes == nil {
+			return nil, errors.New("DescribeInstanceTypesWithOptions failed to return any instance types")
+		}
+
+		if len(resp.Body.InstanceTypes.InstanceType) == 0 {
+			break
```

Comment:
By actually calling it, I found that `NextToken == "" ` can also determine the end.